providers/runtime.go

```diff
@@ -250,6 +250,14 @@ func (r *Runtime) Connect(req *plugin.ConnectReq) error {
 		}
 	}
 
+	// We have a special case here, where we loose the platform information
+	// when doing container-image discovery for k8s.
+	// This way we make sure to preserve the platform information across plugins.
+	pf := r.Provider.Connection.Asset.Platform
+	if req.Asset.Platform == nil {
```

Comment:
I don't want to over-complicate this PR but isn't it better to figure out where we need to explicitly store the response of `Connect`? Otherwise in the future some other field may be missing and we need to add another condition here

Reply:
Another solution: https://github.com/mondoohq/cnspec/pull/981

Reply:
This is where we get the response from the plugin:
https://github.com/mondoohq/cnquery/blob/main/providers/runtime.go#L223

```
r.Provider.Connection, r.Provider.ConnectionError = r.Provider.Instance.Plugin.Connect(req, &callbacks)
```

But we do not return it from there, so we have to use it explicitly. At least this is my current understanding.